modules/telemetry/telemetry/src/metric_record.cpp

```diff
@@ -115,16 +116,20 @@ class MetricRecorder {
 private:
   absl::Mutex sink_mutex_;
   std::vector<std::unique_ptr<IMetricSink>> sinks_ ABSL_GUARDED_BY(sink_mutex_);
-  containers::BlockingQueue<Metric> entries_;
+  containers::BlockingQueue<heph::UniqueFunction<Metric()>> entries_;
   std::future<void> message_process_future_;
 };
 
 void registerMetricSink(std::unique_ptr<IMetricSink> sink) {
   MetricRecorder::registerSink(std::move(sink));
 }
 
-void record(const Metric& metric) {
-  MetricRecorder::record(metric);
+void record(heph::UniqueFunction<Metric()> metric) {
```

Comment:
since its unique isn't it gonnda be a && always? in that case you can also make the interface narrower directly

Reply:
good point.